swagger.yaml

```diff
@@ -0,0 +1,249 @@
+        404:
+          description: page not found
+          schema:
+            $ref: "#/definitions/Error"
+        405:
+          description: Invalid input
+          schema:
+            $ref: "#/definitions/Error"
+        default:
+          description: Unexpected error
+          schema:
+            $ref: "#/definitions/Error"
+
+    post:
+      tags:
+      - tweets
+      description: "post a new tweet"
+      produces:
+      - application/json
+
+      parameters:
+        - name: body
+          in: body
+          description: "tweet will be created and posted"
+          required: true
+          schema:
+            $ref: "#/definitions/tweetsData"
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/tweetsData"
+        204:
+          description: No content
+          schema:
+            $ref: "#/definitions/Error"
+        400:
+          description: Bad request
+          schema:
+            $ref: "#/definitions/Error"
+        404:
+          description: page not found
+          schema:
+            $ref: "#/definitions/Error"
+        405:
+          description: Invalid input
+          schema:
+            $ref: "#/definitions/Error"
+        default:
+          description: Unexpected error
+          schema:
+            $ref: "#/definitions/Error"
+
+  /tweets/{tweetId}:
+
+    get:
+      tags:
+        - tweets
+      description: "Find a tweet by tweet ID"
+      produces:
+      - application/json
+      parameters:
+        - name: tweetId
+          in: path
+          required: true
+          type: integer
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/tweetsData"
+        204:
+          description: No content
+          schema:
+            $ref: "#/definitions/Error"
+        400:
+          description: Bad request
+          schema:
+            $ref: "#/definitions/Error"
+        404:
+          description: page not found
+          schema:
+            $ref: "#/definitions/Error"
+        405:
+          description: Invalid input
+          schema:
+            $ref: "#/definitions/Error"
+        default:
+          description: Unexpected error
+          schema:
+            $ref: "#/definitions/Error"
+
+    put:
+      tags:
+      - tweets
+      description: "update a tweet"
+      produces:
+      - application/json
+
+      parameters:
+        - name: tweetId
+          in: path
+          description: "update the tweet"
+          required: true
+          type: integer
+
+        - name: body
+          in: body
+          description: "tweet content to be updated"
+          required: true
+          schema:
+            $ref: "#/definitions/tweetsData"
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/tweetsData"
+        204:
```

Comment:
Like the POST method above, when do you wanna to return a `204` for `PUT` method?